cirq/ops/common_channels.py

```diff
@@ -50,90 +60,156 @@ def __init__(self, p_x: float, p_y: float, p_z: float) -> None:
         """
-
-        self._p_x = value.validate_probability(p_x, 'p_x')
-        self._p_y = value.validate_probability(p_y, 'p_y')
-        self._p_z = value.validate_probability(p_z, 'p_z')
-        self._p_i = 1 - value.validate_probability(p_x + p_y + p_z,
-                                                   'p_x + p_y + p_z')
+        assert ('pis' in kwargs or 'p_x' in kwargs or 'p_y' in kwargs or
+                'p_z' in kwargs or len(args) > 0)
+        if 'pis' in kwargs:
+            assert 'num_qubits' in kwargs
+            self._num_qubits = kwargs['num_qubits']
+            self._pis = kwargs['pis']
+        else:
+            if len(args) > 0:
+                p_x = args[0]
+                p_y = args[1] if len(args) > 1 else 0.0
+                p_z = args[2] if len(args) > 2 else 0.0
+            else:
+                p_x = kwargs.get('p_x', 0.0)
+                p_y = kwargs.get('p_y', 0.0)
+                p_z = kwargs.get('p_z', 0.0)
+            self._num_qubits = 1
+
+            p_x = value.validate_probability(p_x, 'p_x')
+            p_y = value.validate_probability(p_y, 'p_y')
+            p_z = value.validate_probability(p_z, 'p_z')
+            p_i = 1 - value.validate_probability(p_x + p_y + p_z,
+                                                 'p_x + p_y + p_z')
+
+            self._pis = {'I': p_i, 'X': p_x, 'Y': p_y, 'Z': p_z}
 
     def _mixture_(self) -> Sequence[Tuple[float, np.ndarray]]:
-        return ((self._p_i, protocols.unitary(identity.I)),
-                (self._p_x, protocols.unitary(pauli_gates.X)),
-                (self._p_y, protocols.unitary(pauli_gates.Y)),
-                (self._p_z, protocols.unitary(pauli_gates.Z)))
+        Pis = []
+        for pauli in self._pis.keys():
+            Pi = np.identity(1)
+            for gate in pauli:
+                if gate == 'I':
+                    Pi = np.kron(Pi, protocols.unitary(identity.I))
+                elif gate == 'X':
+                    Pi = np.kron(Pi, protocols.unitary(pauli_gates.X))
+                elif gate == 'Y':
+                    Pi = np.kron(Pi, protocols.unitary(pauli_gates.Y))
+                elif gate == 'Z':
+                    Pi = np.kron(Pi, protocols.unitary(pauli_gates.Z))
+            Pis.append(Pi)
+        return tuple(zip(self._pis.values(), Pis))
+
+    def _num_qubits_(self) -> int:
+        return self._num_qubits
 
     def _has_mixture_(self) -> bool:
         return True
 
     def _value_equality_values_(self):
-        return self._p_x, self._p_y, self._p_z
+        return self._num_qubits, hash(tuple(sorted(self._pis.items())))
 
     def __repr__(self) -> str:
-        return 'cirq.asymmetric_depolarize(p_x={!r},p_y={!r},p_z={!r})'.format(
-            self._p_x, self._p_y, self._p_z
-        )
+        return (
+            'cirq.asymmetric_depolarize' +
+            '(num_qubits={!r},pis={!r})'.format(self._num_qubits, self._pis))
```

Comment:
Let's remove "num_qubits={!r}," if num_qubits is one (the default).

Reply:
Done.